builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -38,13 +38,22 @@ func resourceAwsS3Bucket() *schema.Resource {
 					Schema: map[string]*schema.Schema{
 						"index_document": &schema.Schema{
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
 						},
 
 						"error_document": &schema.Schema{
 							Type:     schema.TypeString,
 							Optional: true,
 						},
+
+						"redirect_all_requests_to": &schema.Schema{
+							Type: schema.TypeString,
+							ConflictsWith: []string{
+								"website.0.index_document",
+								"website.0.error_document",
+							},
```

Comment:
Being the original author of `ConflictsWith`, I know about this limitation, but I was finding it hard to create *any nested schema* (can't remember what was the problem exactly though), putting aside `ConflictsWith`.